glean-core/python/setup.py

```diff
@@ -116,20 +92,24 @@ def has_ext_modules(self):
 # simple that only handles the cases we need.
 class bdist_wheel(wheel.bdist_wheel.bdist_wheel):
     def get_tag(self):
-        if platform == "linux":
-            return ("cp36", "abi3", "linux_x86_64")
-        elif platform == "darwin":
-            return ("cp36", "abi3", "macosx_10_7_x86_64")
-        elif platform == "windows":
-            if mingw_arch == "i686":
-                return ("py3", "none", "win32")
-            elif mingw_arch == "x86_64":
-                return ("py3", "none", "win_amd64")
+        cpu, _, __ = target.partition("-")
+        if "-linux" in target:
+            plat_name = f"linux_{cpu}"
+        elif "-darwin" in target:
+            compat = macos_compat(target).replace(".", "_")
+            plat_name = f"macosx_{compat}_{cpu}"
+        elif "-windows" in target:
+            if cpu == "i686":
+                plat_name = "win32"
+            elif cpu == "x86_64":
+                plat_name = "win_amd64"
             else:
                 raise ValueError("Unsupported Windows platform")
         else:
             # Keep local wheel build on BSD/etc. working
-            return super().get_tag()
+            _, __, plat_name = super().get_tag()
+
+        return ("cp36", "abi3", plat_name)
```

Comment:
Have you confirmed this works with Windows wheels?  The pattern matching by pip is slightly different there, hence the weird hard coding of all of the supported options that was here before.  (Related, let's update the comment above this function if this new approach *does* work).

Reply:
It worked for me locally, but it also obviously works more generally, since Windows python tests on CI use the wheels.

Reply:
Note the comment above get_tag is still relevant as-is.

Reply:
I just reminded myself of the details of the issue (which we should probably put in a comment so it isn't lost again).

Windows Python prior to 3.7 does not have a stable ABI, so therefore doesn't support the `abi3` tag.  Therefore, it won't "match" that wheel and it falls back to installing from source.  You can test this locally by putting the wheels in a local directory and forcing `pip` to look there instead of PyPI with the `-f directory` option.  It won't work with the wheel named as this PR does it.

Of course, we don't actually use the Python3 ABI since we use CFFI, so the best way to make a wheel that will work on Windows Python 3.6 and up is to mark it with `py3-none` rather than `cp36-abi3`.

Reply:
I'm not sure this is the full story. Because while installing a fresh Windows Python 3.6 and running `pip install glean_sdk-35.0.0-cp36-abi3-win_amd64.whl` doesn't work, pip tells you it's version 18.1 and that you should upgrade to 21.0.1, and after doing that, it works. Trying a few versions, it started working with pip 20.0.1.

Reply:
Installing directly from a file path always works.  What doesn't is the pattern matching engine. So you need to do:

```
pip install -f directory_containing_wheels glean_sdk
```

to match what would happen in normal situations when installing from pypi.

This doesn't work for me on Python 3.6, even when upgrading pip.

Reply:
> Installing directly from a file path always works

That's not my experience:
```
# ./AppData/Local/Programs/Python/Python36/python -m pip install glean_sdk-35.0.0-cp36-abi3-win_amd64.whl
glean_sdk-35.0.0-cp36-abi3-win_amd64.whl is not a supported wheel on this platform.
You are using pip version 18.1, however version 21.0.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
```

> This doesn't work for me on Python 3.6, even when upgrading pip.

That doesn't look to be true (after upgrading to pip 21.0.1):
```
# ./AppData/Local/Programs/Python/Python36/python -m pip install -f . glean_sdk
Looking in links: .
Processing c:\users\glandium\glean_sdk-35.0.0-cp36-abi3-win_amd64.whl
(...)
```

Reply:
Thanks for that.  I think I wasn't upgrading `pip` correctly.

So then I think the only thing to confirm is that we aren't held back to a particular version of `pip` by how things are vendored in m-c.  Assuming that checks out, :+1: from me.

Reply:
Thank you for making me look. Things are grim, unfortunately, as the common way to install and upgrade moz-phab on Windows uses pip 19.3 :(